### PR TITLE
Feature/added symbol-autocomplete to library (#2003)

### PR DESCRIPTION
**Pull request by aGhosh02**

### CODE REVIEW

File: libs/ui/src/lib/abstract-mat-form-field.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+export abstract class AbstractMatFormField<T>
+  implements DoCheck, OnDestroy, ControlValueAccessor, MatFormFieldControl<T>
+{
+  private static nextId: number = 0;
+  @HostBinding()
+  public id: string = `${this.controlType}-${AbstractMatFormField.nextId++}`;
+  @HostBinding('attr.aria-describedBy')
+  public describedBy: string = '';
+  public focused = false;
+  readonly autofilled: boolean;
```
Comment (dtslvr) [inline libs/ui/src/lib/abstract-mat-form-field.ts:29]: Can you please add `private` / `protected` / `public` to all class variables?

File: libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  takeUntil
+} from 'rxjs/operators';
+
+import { DataService } from '../../../../../apps/client/src/app/services/data.service';
+import { AbstractMatFormField } from '../abstract-mat-form-field';
+
+@Component({
+  host: {
+    '[id]': 'id',
+    '[attr.aria-describedBy]': 'describedBy'
```
Comment (dtslvr) [inline libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.ts:32]: Move up ⬆️  to order alphabetically

File: libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      useExisting: SymbolAutocompleteComponent
+    }
+  ]
+})
+export class SymbolAutocompleteComponent
+  extends AbstractMatFormField<LookupItem>
+  implements OnInit, OnDestroy
+{
+  public control = new FormControl();
+  filteredLookupItemsObservable: Observable<LookupItem[]> = of([]);
```
Comment (dtslvr) [inline libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.ts:50]: `public`?

File: libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+  public ngOnInit(): void {
+    super.required = this.ngControl.control?.hasValidator(Validators.required);
+
+    this.filteredLookupItemsObservable = this.control.valueChanges.pipe(
+      debounceTime(400),
+      distinctUntilChanged(),
+      switchMap((query: string) => {
+        if (isString(query) && query.length > 1) {
+          const response = this.dataService
```
Comment (dtslvr) [inline libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.ts:86]:
```
return this.dataService
.fetchSymbols(query)
.pipe(takeUntil(this.unsubscribeSubject))
.subscribe((filteredLookupItems) => {
  this.filteredLookupItems = filteredLookupItems;
});
```

File: libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.module.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import { RouterModule } from '@angular/router';
+import { GfSymbolModule } from '@ghostfolio/client/pipes/symbol/symbol.module';
+import { SymbolAutocompleteComponent } from '@ghostfolio/ui/symbol-autocomplete/symbol-autocomplete.component';
+
+@NgModule({
+  declarations: [SymbolAutocompleteComponent],
+  exports: [SymbolAutocompleteComponent],
+  imports: [
+    CommonModule,
+    RouterModule,
```
Comment (dtslvr) [inline libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.module.ts:17]: Do you need the `RouterModule`?

File: libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.module.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import { MatInputModule } from '@angular/material/input';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { RouterModule } from '@angular/router';
+import { GfSymbolModule } from '@ghostfolio/client/pipes/symbol/symbol.module';
+import { SymbolAutocompleteComponent } from '@ghostfolio/ui/symbol-autocomplete/symbol-autocomplete.component';
+
+@NgModule({
+  declarations: [SymbolAutocompleteComponent],
+  exports: [SymbolAutocompleteComponent],
+  imports: [
```
Comment (dtslvr) [inline libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.module.ts:15]: Order imports alphabetically

**Review by dtslvr — CHANGES_REQUESTED** [review]
- [x] When you open the create activity dialog, the symbol search is triggered and a request is sent to the backend: `/api/v1/symbol/undefined/undefined`
- [x] When you type anything in the symbol search field, always two requests are sent to the backend. Can you check that, too? This was also an issue before your change I think.
- [x] When you edit an existing activity, the symbol search is disabled. It's displayed with grey text but you can still search and edit the value

File: libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.ts (hunk 3 above)
Comment (aGhosh02) [reply libs/ui/src/lib/symbol-autocomplete/symbol-autocomplete.component.ts:52]: Neds to be public as its being used in html

**Review by dtslvr — APPROVED** [review]
Very cool improvement @aGhosh02! Especially the fixes with the multiple requests 🥳 

I have made some minor refactoring and improved the loading spinner.

**Pull request merged**